core/python/timestreamextensions.py

```diff
@@ -41,61 +42,92 @@ def timestreamastype(a, dtype):
     return out
 G3Timestream.astype = timestreamastype
 
+# XXX consider replacing all of this with the numpy __array_ufunc__ machinery
+
 # Provide all the numpy binary arithmetic operators, first non-in-place, then
 # in-place, with slightly different semantics
 def numpybinarywrap(a, b, op):
-    if isinstance(b, G3Timestream):
-        a._assert_congruence(b)
-    out = G3Timestream(op(numpy.asarray(a), numpy.asarray(b)))
-    out.units = a.units
-    out.start = a.start
-    out.stop = a.stop
-    return out
-
-for x in ['__add__', '__and__', '__div__', '__divmod__', '__floordiv__', 
-          '__mul__', '__neg__', '__or__', '__pow__', '__sub__', '__radd__',
-          '__rdiv__', '__rdivmod__', '__rmod__', '__rmul__', '__rpow__',
-          '__rsub__', '__rtruediv__', '__truediv__']:
-    if x in numpy.ndarray.__dict__:
-        setattr(G3Timestream, x, 
-                lambda a, b, op=numpy.ndarray.__dict__[x]: numpybinarywrap(a, b, op))
+    is_g3 = 'G3' in a.__class__.__name__ or 'G3' in b.__class__.__name__
```

Comment:
Maybe check if it is a kind of G3FrameObject instead? I think that's the actual test you want to do here.

For example,
```suggestion
    is_g3 = issubclass(a, core.G3FrameObject) or issubclass(b, core.G3FrameObject)
```